databallpy/utils/game_utils.py

```diff
@@ -120,3 +121,157 @@ def create_event_attributes_dataframe(
         attr: [getattr(event, attr) for event in events.values()] for attr in attributes
     }
     return pd.DataFrame(res_dict)
+
+
+def _add_starter_information(
+    metadata: Metadata,
+    tracking_data: pd.DataFrame | None = None,
+    event_data: pd.DataFrame | None = None,
+) -> Metadata:
+    """Function to add starter information to metadata when not provided by the data source.
+
+    This function will only add starter information when it's missing (all None or all False).
+    It prioritizes tracking data if available, otherwise uses event data.
+
+    Args:
+        metadata (Metadata): The metadata object containing player information
+        tracking_data (pd.DataFrame | None): Optional tracking data to determine starters
+            from first 22 players (11 per team) with non-null data. Defaults to None.
+        event_data (pd.DataFrame | None): Optional event data to determine starters
+            from substitute events and player participation. Defaults to None.
+
+    Returns:
+        Metadata: Updated metadata with starter information added
+    """
+    # Check if starter information already exists and has meaningful values
+    home_has_starters = (
+        "starter" in metadata.home_players.columns
+        and metadata.home_players["starter"].notna().any()
+        and metadata.home_players["starter"].any()
+    )
+    away_has_starters = (
+        "starter" in metadata.away_players.columns
+        and metadata.away_players["starter"].notna().any()
+        and metadata.away_players["starter"].any()
+    )
+
+    if home_has_starters and away_has_starters:
+        # Starter information already exists for both teams, no need to add
+        return metadata
+
+    # Initialize starter column if it doesn't exist
+    if "starter" not in metadata.home_players.columns:
+        metadata.home_players["starter"] = False
+    if "starter" not in metadata.away_players.columns:
+        metadata.away_players["starter"] = False
+
+    # Try to use tracking data first
+    if tracking_data is not None and not tracking_data.empty:
+        _add_starters_from_tracking_data(metadata, tracking_data)
+    elif event_data is not None and not event_data.empty:
+        _add_starters_from_event_data(metadata, event_data)
+
+    return metadata
+
+
+def _add_starters_from_tracking_data(
```

Comment:
Instead of assuming the first index has all players, look for the first frame in the first 5 minutes of the game where there is data of 22 players. Somethimes the tracking data capture starts before players are visible in the data.